### PR TITLE
Created Somatic-PON-QC workflow

### PR DESCRIPTION
**Pull request by skanwal**
# Completed steps

- [x] Merged #80 
- [x] Merged #82 

# Next steps
- [x] Follow-up and register `wfr.e0cccb29cf44482182507bd33579270c`
- [x] Merge #83 
- [x] Merge #84 
- [x] Merge main into this branch
- [x] Covert to review ready PR

cc - @scwatts (in case you would like to stay updated on this workflow)

### CODE REVIEW

**Comment by skanwal** [comment]
@alexiswl  - ready for review

File: workflows/dragen-pon-qc/3.9.3/dragen-pon-qc__3.9.3.cwl [hunk 1]
```diff
@@ -0,0 +1,912 @@
...
+      output file
+    type: string?
+
+steps:
+  # Run dragen somatic tool using normal fastqs
+  run_dragen_somatic_step:
+    label: run dragen somatic step
+    doc: |
+      Runs the dragen somatic workflow on the FPGA.
+      Takes in a normal and tumor fastq list and corresponding mount paths from the predefined_mount_paths.
```
Comment (alexiswl) [inline workflows/dragen-pon-qc/3.9.3/dragen-pon-qc__3.9.3.cwl:674]: Doc slightly out of date. Takes in fastq list rows now and generates csv in the tool

File: workflows/dragen-pon-qc/3.9.3/dragen-pon-qc__3.9.3.cwl [hunk 2]
```diff
@@ -0,0 +1,912 @@
...
+  # --enable-duplicate-marking to mark duplicate reads at the same time
+  # --enable-sv to enable the structural variant calling step.
+  enable_map_align_output:
+    label: enable map align output
+    doc: |
+      Enables saving the output from the
+      map/align stage. Default is true when only
+      running map/align. Default is false if
+      running the variant caller.
+    type: boolean?
```
Comment (alexiswl) [inline workflows/dragen-pon-qc/3.9.3/dragen-pon-qc__3.9.3.cwl:75]: Is it appropriate to leave this as an input to the user for this workflow? Should this be set to something by default?

**Comment by alexiswl** [comment]
Looks like a few items from config/run.yaml have been deleted. Could this have been an issue with merging main into this branch?

File: workflows/dragen-pon-qc/3.9.3/dragen-pon-qc__3.9.3.cwl [hunk 3]
```diff
@@ -0,0 +1,912 @@
...
+      tumor_fastq_list_rows:
+        source: tumor_fastq_list_rows
+      reference_tar:
+        source: reference_tar
+      output_directory:
+        source: output_directory
+      output_file_prefix:
+        source: output_file_prefix
+      enable_map_align_output:
+        source: enable_map_align_output
```
Comment (alexiswl) [inline workflows/dragen-pon-qc/3.9.3/dragen-pon-qc__3.9.3.cwl:688]: As discussed in meeting, change from `source: enable_map_align_output` to 

```
valueFrom: |
  ${ return true; }
```

File: workflows/dragen-pon-qc/3.9.3/dragen-pon-qc__3.9.3.cwl (hunk 2 above)
Comment (alexiswl) [reply workflows/dragen-pon-qc/3.9.3/dragen-pon-qc__3.9.3.cwl:75]: As discussed, if set to false will fail, just remove this, and set to true in step level

File: workflows/dragen-pon-qc/3.9.3/dragen-pon-qc__3.9.3.cwl (hunk 3 above)
Comment (skanwal) [reply workflows/dragen-pon-qc/3.9.3/dragen-pon-qc__3.9.3.cwl]: fixed in https://github.com/umccr/cwl-ica/pull/85/commits/f501c4effabbf62bb55084de6998dd12789dc882

File: workflows/dragen-pon-qc/3.9.3/dragen-pon-qc__3.9.3.cwl (hunk 2 above)
Comment (skanwal) [reply workflows/dragen-pon-qc/3.9.3/dragen-pon-qc__3.9.3.cwl]: fixed in https://github.com/umccr/cwl-ica/pull/85/commits/f501c4effabbf62bb55084de6998dd12789dc882

File: workflows/dragen-pon-qc/3.9.3/dragen-pon-qc__3.9.3.cwl (hunk 1 above)
Comment (skanwal) [reply workflows/dragen-pon-qc/3.9.3/dragen-pon-qc__3.9.3.cwl]: fixed in https://github.com/umccr/cwl-ica/pull/85/commits/f501c4effabbf62bb55084de6998dd12789dc882

**Comment by skanwal** [comment]
> Looks like a few items from config/run.yaml have been deleted. Could this have been an issue with merging main into this branch?

updated!

**Review by alexiswl — APPROVED** [review]

**Pull request merged**